core/src/main/java/org/apache/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -94,7 +94,7 @@ public List<RoleEntity> listRolesByUserIdent(NameIdentifier userIdent) {
         .map(
             po ->
                 POConverters.fromRolePO(
-                    po, Collections.emptyList(), AuthorizationUtils.ofRoleNamespace(metalake)))
+                    po, listSecurableObjects(po), AuthorizationUtils.ofRoleNamespace(metalake)))
```

Comment:
You can't do this. This will cause performance issues.

Reply:
fixed